### PR TITLE
Remove timeout in async_test for miscellaneous tests

### PR DESCRIPTION
**Pull request by qiuzhong**
Remove all the timeout in `async_test` for miscellaneous tests
except html, IndexedDB, websockets, mediacapture-streams and cookies.
Affected tests: 109
Before: Pass: 108, Failed: 1
After: Pass: 108, Failed: 1
Related: #11120

### CODE REVIEW

**Review by foolip — APPROVED** [review]

File: resources/test/tests/functional/api-tests-1.html [hunk 1]
```diff
@@ -193,7 +193,7 @@ <h1>Sample HTML5 API Tests</h1>
     async_test("test should timeout (fail) with a custom set timeout value of 1 second",
                {timeout:1000}).step(function(){});
 
-    async_test("async test that is never started, should have status Not Run", {timeout:1000});
+    async_test("async test that is never started, should have status Not Run");
```
Comment (jdm) [inline resources/test/tests/functional/api-tests-1.html]: This test will need to be updated (probably by removing it), since it's specifically about timing out after one second. There's a failing travisci job because of this change.

**Review by jdm — DISMISSED** [review]

**Comment by jdm** [comment]
The chrome travisci job reports that one of these tests is now too slow, apparently:
```
NFO:/home/travis/build/web-platform-tests/wpt/tools/ci:Starting tests
ERROR:/home/travis/build/web-platform-tests/wpt/tools/ci:## Slow tests ##
ERROR:/home/travis/build/web-platform-tests/wpt/tools/ci:|                         Test                         | Result | Longest duration (ms) | Timeout (ms) |
ERROR:/home/travis/build/web-platform-tests/wpt/tools/ci:|------------------------------------------------------|--------|-----------------------|--------------|
ERROR:/home/travis/build/web-platform-tests/wpt/tools/ci:| `/geolocation-API/getCurrentPosition_IDL.https.html` | `OK`   | `10427`               | `10000`      |
```

**Comment by foolip** [comment]
@qiuzhong, I'd recommend backing out the changes to just those tests, merging this PR, and then fixing those separately.

File: resources/test/tests/functional/api-tests-1.html (hunk 1 above)
Comment (qiuzhong) [reply resources/test/tests/functional/api-tests-1.html]: @jdm, thanks for your remark. You're right, I'll remove it and other similar tests about timing out.

**Comment by qiuzhong** [comment]
@foolip , I reverted the failed test to make the CI pass.

File: resources/test/tests/functional/api-tests-1.html [hunk 2]
```diff
@@ -190,12 +190,6 @@ <h1>Sample HTML5 API Tests</h1>
 
     async_test("test should timeout (fail) with the default of 2 seconds").step(function(){});
 
-    async_test("test should timeout (fail) with a custom set timeout value of 1 second",
```
Comment (foolip) [inline resources/test/tests/functional/api-tests-1.html]: Can you revert these changes? We should delete the tests when the functionality itself is removed eventually.

Comment (qiuzhong) [reply resources/test/tests/functional/api-tests-1.html]: Reverted.

**Comment by foolip** [comment]
@qiuzhong looks like you force pushed which reverted a whitespace revert that I'd done. Can you push again with all of the intended changes?

**Comment by qiuzhong** [comment]
@foolip , I've removed the whitespace this time. It's not good practice to use `--force` for `git push` and I'll avoid using it.

**Comment by foolip** [comment]
> It's not good practice to use `--force` for `git push` and I'll avoid using it.

I like rewriting history myself, but try `git push --force-with-lease` to avoid mishaps. I've made it an alias `git pushf` since it really should be the default.

**Comment by foolip** [comment]
Looks like I got that from https://stackoverflow.com/questions/30542491/push-force-with-lease-by-default.

**Comment by foolip** [comment]
Changes to resources/test/tests/functional/api-tests-1.html are still in this PR. @qiuzhong, can you revert those?

**Comment by qiuzhong** [comment]
@foolip, I'm a little confused. The removed test in resources/test/tests/functional/api-tests-1.html is specially about timing out. Since `timeout` is removed from `async_test`, @jdm would suggest to [remove it if possible](https://github.com/web-platform-tests/wpt/pull/14094#discussion_r234192994) in the review above. You suggest to revert it now and then fix it separately?

**Comment by foolip** [comment]
@qiuzhong I would expect those tests to be removed when the functionality itself is removed from testharness.js, after all of the "real" tests have been updated in PRs like this one.

**Comment by foolip** [comment]
So yes, unless I'm missing what @jdm had in mind, reverting any changes in resources/ and doing them separately is what I'd suggest.

**Comment by qiuzhong** [comment]
@foolip, all changes in resources/ are reverted now.

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
Before/after results:
https://wpt.fyi/results/?label=taskcluster&product=chrome%405438f72ea0&product=chrome%40565cc2b9a4&diff
https://wpt.fyi/results/?label=taskcluster&product=firefox%405438f72ea0&product=firefox%40565cc2b9a4&diff

There are apparent regressions, but none in directories touched by PRs, so probably flakiness? @lukebjerring, does this seem like a normal amount of flakiness noise? Wasn't there a recent duplicate run you checked where there were literally no differences?

**Comment by lukebjerring** [comment]
Not for a full run - it was duplicate runs for some affected tests.
https://staging.wpt.fyi/results/?product=firefox&sha=2b35e61606&max-count=30
